SampleApplications/Samples/XamarinClient/XamarinClient/SampleClient.cs

```diff
@@ -0,0 +1,469 @@
+                TransportConfigurations = new TransportConfigurationCollection(),
+                TransportQuotas = new TransportQuotas { OperationTimeout = 15000 },
+                ClientConfiguration = new ClientConfiguration { DefaultSessionTimeout = 60000 }
+            };
+
+            switch (Device.RuntimePlatform)
+            {
+                case "Android":
+                    config.ApplicationName = "UA Core Sample Client Android";
+                    break;
+                case "UWP":
+                    config.ApplicationName = "UA Core Sample Client UWP";
+                    break;
+                case "iOS":
+                    config.ApplicationName = "UA Core Sample Client IOS";
+                    break;
+            }
+      
+            await config.Validate(ApplicationType.Client);
+
+            haveAppCertificate = config.SecurityConfiguration.ApplicationCertificate.Certificate != null;
+
+            if (!haveAppCertificate)
+            {
+                info.LabelText = "    INFO: Creating new application certificate: " + config.ApplicationName;
+
+                X509Certificate2 certificate = CertificateFactory.CreateCertificate(
+                    config.SecurityConfiguration.ApplicationCertificate.StoreType,
+                    config.SecurityConfiguration.ApplicationCertificate.StorePath,
+                    null,
+                    config.ApplicationUri,
+                    config.ApplicationName,
+                    config.SecurityConfiguration.ApplicationCertificate.SubjectName,
+                    null,
+                    CertificateFactory.defaultKeySize,
+                    DateTime.UtcNow - TimeSpan.FromDays(1),
+                    CertificateFactory.defaultLifeTime,
+                    CertificateFactory.defaultHashSize,
+                    false,
+                    null,
+                    null);
+
+                config.SecurityConfiguration.ApplicationCertificate.Certificate = certificate;
+            }
+
+            haveAppCertificate = config.SecurityConfiguration.ApplicationCertificate.Certificate != null;
+
+            if (haveAppCertificate)
+            {
+                config.ApplicationUri = Utils.GetApplicationUriFromCertificate(config.SecurityConfiguration.ApplicationCertificate.Certificate);
+
+                if (config.SecurityConfiguration.AutoAcceptUntrustedCertificates)
+                {
+                    config.CertificateValidator.CertificateValidation += new CertificateValidationEventHandler(CertificateValidator_CertificateValidation);
+                }
+            }
+        }
+
+        public async Task<ConnectionStatus>OpcClient(string endpointURL)
+        {
+            try
+            {
+                Uri endpointURI = new Uri(endpointURL);
+                var endpointCollection = DiscoverEndpoints(config, endpointURI, 10);
+                var selectedEndpoint = SelectUaTcpEndpoint(endpointCollection, haveAppCertificate);
```

Comment:
>SelectUaTcpEndpoint [](start = 39, length = 19)

outdated, should use            var selectedEndpoint = CoreClientUtils.SelectEndpoint(endpointURL, haveAppCertificate, 15000);

Reply:
changed